File: 900/1451B/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main() 
{
    int q;
    cin>>q;

    while(q--) 
    {
        int n, t;
        cin>>n>>t;
        string s;
        cin>>s;

        while(t--) 
        {
            int l_1, r_1;
            cin>>l_1>>r_1;

            int l = l_1 - 1;
            int r = r_1 - 1;

            // Equivalent to Java's substring and indexOf checks
            bool left_match = s.substr(0, l).find(s[l]) != string::npos;
            bool right_match = s.substr(r + 1).find(s[r]) != string::npos;

            if (left_match || right_match)
                cout<<"YES\n";
            else
                cout<<"NO\n";
        }
    }

    return 0;
}

File: 900/1451B/main.java
import java.util.*;
public class main 
{
    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
        int q = sc.nextInt();
 
        while(q-- > 0) 
        {
            int n = sc.nextInt();
            int t = sc.nextInt();
            String s = sc.next();
 
            while(t-- > 0) 
            {
                int l = sc.nextInt() - 1;
                int r = sc.nextInt();
 
                if(s.substring(0, l).indexOf(s.charAt(l)) >= 0 || s.substring(r).indexOf(s.charAt(r - 1)) >= 0)
                    System.out.println("YES");
                else
                    System.out.println("NO");
            }
        }
    }
}
